### PR TITLE
Add system.dropped_tables_parts table

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Table system.dropped_tables_parts contains parts of system.dropped_tables tables (dropped but not yet removed tables)

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit f5fb169c65a839155ce88ad9a511a88835398afa with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57555/f5fb169c65a839155ce88ad9a511a88835398afa/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>❌ failure</td></tr>
<tbody>
</table>

File: src/Storages/System/StorageSystemDroppedTablesParts.cpp [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+#include <DataTypes/DataTypeUUID.h>
+#include <Parsers/queryToString.h>
+#include <Interpreters/TransactionVersionMetadata.h>
+#include <Interpreters/Context.h>
+
+
+namespace
+{
+
+std::string_view getRemovalStateDescription(DB::DataPartRemovalState state)
```
Comment (serxa) [inline src/Storages/System/StorageSystemDroppedTablesParts.cpp:21]: Maybe move it closer to a definition of the enum itself to avoid code duplication?

**Review by serxa — COMMENTED** [review]
It feels like it should be always the same as `StorageSystemParts`. Maybe we can move common code to some base class. `StorageSystemPartsBase` feels like a good candidate. But maybe introduce new one, because it is used in a lot of places already.

**Review by serxa — APPROVED** [review]

**Comment by antonio2368** [comment]
This breaks some stateless tests using DatabaseOrdinary
I suspect it's something with this line https://github.com/ClickHouse/ClickHouse/pull/57555/files#diff-6b38d363b27bedd99c449eb3b8b4a2bb3b271d4313659f2e050bc3cf54e1efe2R154 
I think Ordinary databases don't have unique UUID for tables

**Pull request merged**